### ISSUE TITLE
Support sbt 1 in semanticdb-scalac

### ISSUE THREAD

**Issue by olafurpg** [scalameta/scalameta#1157]
Now that sbt 1.0 is on scala 2.12 we should be able to deprecate https://github.com/scalameta/semanticdb-sbt in favor of semanticdb-scalac.

To do this we need to port some customizations in semanticdb-sbt

- [ ] detect SbtGlobal and set dialect to Sbt1
- [ ] collect all Document for same *.sbt file, don't persist right away
- [ ] merge List[Document] from the same *.sbt file into Document https://github.com/scalameta/semanticdb-sbt/blob/189ae4e6e6031c80d1ed0f1f0eb91a59efab1083/sbthost/runtime/src/main/scala/scala/meta/semanticdb/SemanticdbSbt.scala#L30-L104 We can do this inside the compiler plugin instead of at runtime like we do for sbt 0.13
- [ ] persist merged Document after typer
- [ ] test pipeline for .sbt files

Sbt 1.1 is estimated to be out in ~3 weeks with the new slash syntax. @dwijnand has been working on a rewrite from `a in b` to `a / b` that works for regular *.scala sources but is blocked by this issue to make the rewrite work for sbt sources

**Comment by olafurpg** [comment]
I'm afraid I will not have time to work on this issue. If someone is motivated to get semanticdb support working for *.sbt files then the checkboxes above are good starting points.

**Comment by wpopielarski** [comment]
hello @olafurpg 

not sure but I can try to do this if still needed

Sincerely

**Comment by olafurpg** [comment]
Hi @wpopielarski The checklist is still relevant. The interesting parts of the semanticdb-sbt compiler plugin is maybe only 40 loc, see https://github.com/scalameta/semanticdb-sbt/blob/20e56dd904e958adba860407d147e634dd3976ee/sbthost/nsc/src/main/scala/scala/meta/internal/sbthost/SbthostPipeline.scala The traversal itself is no longer relevant, since semanticdb-scalac has a more sophisticated implementation of that same logic in DocumentOps. The rest of the code in the same package is copy-pasted from semanticdb-scalac and can be discarded.

The trickiest bit is getting the persistence correctly within SbtGlobal since we need to persist at the end of the phase where all compilation units for each .sbt file have been handled. It could maybe happen here https://github.com/scalameta/scalameta/blob/d1e6104d1b36419e89a7fac650993455ebbf8b7b/scalameta/semanticdb-scalac-core/src/main/scala/scala/meta/internal/SemanticdbPipeline.scala#L126

Merging of `List[Document]` into `Document` can probably be used unchanged https://github.com/scalameta/semanticdb-sbt/blob/189ae4e6e6031c80d1ed0f1f0eb91a59efab1083/sbthost/runtime/src/main/scala/scala/meta/semanticdb/SemanticdbSbt.scala#L30-L104

For tests, only the SbtFileTest https://github.com/scalameta/semanticdb-sbt/blob/20e56dd904e958adba860407d147e634dd3976ee/sbthost/tests/src/test/scala/scala/meta/tests/SemanticdbSbtTest.scala#L75 is relevant, it requires a few customizations in build.sbt https://github.com/scalameta/semanticdb-sbt/blob/20e56dd904e958adba860407d147e634dd3976ee/build.sbt#L67-L78

I would be really happy to see this working, since that would mean definition/references could work for example in an IDE! I personally just don't have time to maintain this feature.

**Comment by wpopielarski** [comment]
Hello Olafur,

I'm trying to update SbthostPipeline following your remarks and I have problem with classes coming from org.langmeta.internal.semanticdb.schema and org.langmeta,semanticdb. Assuming that traversers in DocumentOps are valid I understand that I should extend SbthostPileline with scala.meta.internal.semanticdb.DatabaseOps and use toDocument on compilation unit. But there are type conflicts and I'm wonder if I should map org.langmeta,semanticdb.* to org.langmeta.internal.semanticdb.schema.* or just go to the end with org.langmeta,semanticdb.* ?
